File: polygon_text.cpp
#include <iostream>
#include <cstdint>
#include <cstdio>

// Create n edges polygons
using namespace std;
// Using uint8_t use 75% less RAM than unsigned int
void repeat(char character, uint8_t times)
{
    // Write n times a character from 1 to last 8-bits integer
    for (uint8_t time = 1; time <= times; time++)
    {
        cout << character;
    }    
}
// Step 1: Define function for draw linear functions
void line(int8_t inclination, int8_t x, int8_t y)
{
    // Plot linear function with inclination and coordenates using 8-bits integers
    if (inclination == 0)
    { // No inclination
        // Line (-)
        repeat('*', 42);
        repeat('\n', 1);
    }
    else if (inclination > 0)
    { // Positive inclination
        // Up-Slash (/)
        for (int8_t times = 10 + y; times >= 1; times--)
        {
           // Make many spaces like say linear function
            repeat(' ', times * x);
          // Plot dot  
            repeat('*', 1);
         // Write newline for plot the next dots   
            repeat('\n', 1);
        }    
    }
    else
    { // Negative inclination
        // Down-Slash (\)
        int8_t last = 1;
        for (int8_t times = 10 + y; times >= 1; times--)
        {
          // Write previous spaces each time make more
            repeat(' ', last * x);
            repeat('*', 1);
         // Write next spaces after dot each time do less   
            repeat(' ', times * x);
            repeat('\n', 1);
            last += 1;
        }    
    }
}

/*
     8-bits integers data (the quarter of RAM) less than 32-bit type int
     use 1 byte instead of 4 bytes.
*/    
int8_t inclination, edges;

int main()
{
// step 2: Get 3 edges or more for draw the right polygon
cout << "Input 3 or more edges: ";
scanf("%i", &edges);

while (edges < 3)
{
	cout << "Again but well: ";
	scanf("%i", &edges);
}
// Step 3: Draw polygon base
    line(0, 0, 0);
// Step 4: Change edges direction for shape up polygon
    inclination = (edges - 2) * 45;
// Step 5: Draw the line in the next position
    line(inclination, 1, 0);    
// Step 6: Draw line with opposite inclination in the next position
    line(-inclination, 1, 1);
// Step 7: Repeat (edges - 3) times steps 5 and 6 with other positions        
    for (int8_t times = 1; times <= (edges - 3); times++)
    {
        line(inclination, times, times + 1);
        line(-inclination, times, times + 2);
    }    
  // Restart the position for don't move terminal prompt by extra spaces  
    repeat('\r', 1);
	return 0;
}

File: polygon_text.py
#!/usr/bin/python3

def repeat(character: str = ' ', times: int = 1) -> str:
    '''
        Show a character (space by default)
        n times (from 1 to the last integer)
    '''
    text: str = ''
    for show in range(1, times + 1):
        text += character
    # Give text part for show right after
    return text
# Step 1: Define function for draw the lines of 2D polygon        
def line(inclination: int, x: int, y: int):
    '''
        Plot linear functions
    '''
    plot = ''
    if inclination > 0:
        # Make up inclination Up-Slash (/)
        time = y + 10
        while(time >= 1):
            # Spaces before dot
            plot += repeat(times = time * x)
            # First dot
            plot += repeat('*', 1)
            plot += '\n'
            # Use less spaces progressive way
            time -= 1
    elif inclination == 0:
      # Constant line (-)
        plot += repeat('*', 42)
        plot += '\n'
    else:
      # Down-Slash (\)
        last = 1
        first = y + 10
        while(first >= 1):
          # Before spaces each time more
            plot += repeat(times = last * x)
            plot += repeat('*', 1)
         # After spaces each time less   
            plot += repeat(times = first * x)
            plot += '\n'
        # Variable feedback for fix spaces
            last += 1
            first -= 1
  # Print text the automatic newline is so dangerous
    print(plot)       
# Step 2: Get 3 or more edges for draw polygon of that lines
edges: int = int(input('Write 3 or more edges: '))

while (edges < 3):
	edges: int = int(input('\n3 or more: '))
# Step 4: Change edges direction for shaping the polygon
inclination: int = (edges - 2) * 45
# Step 5: Draw line with that direction and next position
line(inclination, 1, 0)
# Step 6: Draw line with opposite inclination in next position
line(-inclination, 1, 1)
# Step 7: Repeat (edges - 3) times steps 5 and 6 with others positions
for times in range(1, edges - 2):
    line(inclination, times + 1, times)
    line(-inclination, times + 2, times)
# Step 8: Draw base line	
line(0, 0, 0)	
